### PR TITLE
Flex transformation updates

### PR DESCRIPTION
**Pull request by br648**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Update to accommodate flex tables in transformations.

### CODE REVIEW

File: src/main/java/com/conveyal/datatools/manager/jobs/ProcessSingleFeedJob.java [hunk 1]
```diff
@@ -198,15 +199,22 @@ private String getErrorReasonMessage() {
...
-                feedVersion.validationResult.lastCalendarDate));
-            if (feedVersion.validationResult.errorCount > 0) {
-                message.append(String.format("During validation, we found %s issue(s)",
-                    feedVersion.validationResult.errorCount));
+            ValidationResult validationResult = feedVersion.validationResult;
+            if (validationResult != null) {
+                message.append(String.format("New feed version created for %s (valid from %s - %s). ",
+                    feedSource.name,
+                    feedVersion.validationResult.firstCalendarDate,
+                    feedVersion.validationResult.lastCalendarDate
```
Comment (philip-cline) [inline src/main/java/com/conveyal/datatools/manager/jobs/ProcessSingleFeedJob.java]: Reuse variable?
```suggestion
                    validationResult.firstCalendarDate,
                    validationResult.lastCalendarDate
```

File: src/main/java/com/conveyal/datatools/manager/jobs/ProcessSingleFeedJob.java [hunk 2]
```diff
@@ -198,15 +199,22 @@ private String getErrorReasonMessage() {
...
+                ));
             } else {
-                message.append("The validation check found no issues with this new dataset!");
+                message.append(String.format("New feed version created for %s.", feedSource.name));
+            }
+            if (validationResult != null) {
```
Comment (philip-cline) [inline src/main/java/com/conveyal/datatools/manager/jobs/ProcessSingleFeedJob.java]: Can we nest this condition under the `validationResult != null` check above?

**Review by philip-cline — CHANGES_REQUESTED** [review]
Super clean fix, this is looking awesome! Thanks for setting this up. Just a couple of comments

**Review by philip-cline — APPROVED** [review]

File: src/main/java/com/conveyal/datatools/manager/jobs/ProcessSingleFeedJob.java (hunk 2 above)
Comment (binh-dam-ibigroup) [reply src/main/java/com/conveyal/datatools/manager/jobs/ProcessSingleFeedJob.java]: Phil's comment above was not addressed.

File: src/main/java/com/conveyal/datatools/manager/models/transform/NormalizeFieldTransformation.java [hunk 3]
```diff
@@ -179,6 +179,11 @@ public static List<String> getInvalidSubstitutionPatterns(List<Substitution> sub
 
     @Override
     public void transform(FeedTransformZipTarget zipTarget, MonitorableJob.Status status) {
+        if (Table.LOCATION_GEO_JSON_FILE_NAME.equals(table)) {
+            // It's not possible to select the locations.geojson file from the normalize field transformation list. This
+            // is here in case that changes.
+            throw new UnsupportedOperationException("It is not possible to normalize geo json fields.");
+        }
         String tableName = table + ".txt";
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/datatools/manager/models/transform/NormalizeFieldTransformation.java]: Use `getTableName` defined in `FeedTransformation`.

File: src/test/java/com/conveyal/datatools/manager/jobs/ProcessSingleFeedJobNotificationTest.java [hunk 4]
```diff
@@ -60,7 +60,7 @@ public void canNotifySubscribersOfErrors() {
         validationResult.errorCount = 100;
         f.validationResult = validationResult;
         ProcessSingleFeedJob processSingleFeedJob = new ProcessSingleFeedJob(f, user, true);
-        String expectedResult = String.format("New feed version created for Mock Feed Source (valid from %s - %s). During validation, we found 100 issue(s)",
```
Comment (binh-dam-ibigroup) [inline src/test/java/com/conveyal/datatools/manager/jobs/ProcessSingleFeedJobNotificationTest.java]: Revert this change (tweak the underlying code to that space is restored).

File: src/test/java/com/conveyal/datatools/manager/jobs/ProcessSingleFeedJobNotificationTest.java [hunk 5]
```diff
@@ -75,7 +75,7 @@ public void canNotifySubscribersOfSuccess() {
         validationResult.errorCount = 0;
         f.validationResult = validationResult;
         ProcessSingleFeedJob processSingleFeedJob = new ProcessSingleFeedJob(f, user, true);
-        String expectedResult = String.format("New feed version created for Mock Feed Source (valid from %s - %s). The validation check found no issues with this new dataset!",
```
Comment (binh-dam-ibigroup) [inline src/test/java/com/conveyal/datatools/manager/jobs/ProcessSingleFeedJobNotificationTest.java]: Revert this change (tweak the underlying code to that space is restored).

File: src/test/java/com/conveyal/datatools/manager/jobs/NormalizeFieldTransformJobTest.java [hunk 6]
```diff
@@ -127,7 +145,7 @@ public void canHandleInvalidSubstitutionPatterns() throws IOException {
                 new Substitution("\\Cir\\b", "Circle")
             )
         );
-        initializeFeedSource(transformation);
+        initializeFeedSource(TABLE_NAME, transformation);
```
Comment (binh-dam-ibigroup) [inline src/test/java/com/conveyal/datatools/manager/jobs/NormalizeFieldTransformJobTest.java]: Would it make sense to inline `TABLE_NAME` or `FIELD_NAME`, or move their declarations to this method?

**Review by binh-dam-ibigroup — APPROVED** [review]
Nice cleanup, see the few nice-to-have tweaks. repo.opengeo.org is not responsive at the moment, and I am not able to download dependencies, but I'm going to approve anyway.

**Pull request merged**